test/e2e/scrapeconfig_test.go

```diff
@@ -224,6 +224,87 @@ func testScrapeConfigLifecycle(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func testScrapeConfigLifecycleInDifferentNS(t *testing.T) {
+	skipPrometheusTests(t)
+
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+
+	// The ns where the prometheus CR will reside
+	promns := framework.CreateNamespace(context.Background(), t, testCtx)
+	// The ns where the scrapeConfig will reside
+	scns := framework.CreateNamespace(context.Background(), t, testCtx)
+	framework.SetupPrometheusRBAC(context.Background(), t, testCtx, promns)
+
+	_, err := framework.CreateOrUpdatePrometheusOperator(
+		context.Background(),
+		promns,
+		[]string{promns},
```

Comment:
I believe that the operator doesn't watch for the scrape config namespace.
```suggestion
		[]string{scns},
```

Reply:
I suspected that too, but this too does not yield the expected result.

Reply:
However, I just changed a few things now (including this suggestion), and the test is passing. 
Let me quickly remove/add a few parameters and see what made the tests fail earlier.

Reply:
Well, this is weird.
The test passed, I then changed around a few label selectors, removed them all together and ran the test and it failed. Now, reverting back to the condition where it passed, the test again failed.

Reply:
Findings:
1. The tests failed after letting Operator check the scns because I was looking for targets in scns in my local code.
2. The test failed when I set the label selector to nil value.
3. Setting both the right label selectors for labels and namespace passed the test.